examples/hello-back/Hello.sol

```diff
@@ -0,0 +1,69 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import { IAxelarExecutable } from '@axelar-network/axelar-cgp-solidity/contracts/interfaces/IAxelarExecutable.sol';
+import { IERC20 } from '@axelar-network/axelar-cgp-solidity/contracts/interfaces/IERC20.sol';
+import { IAxelarGasService } from '@axelar-network/axelar-cgp-solidity/contracts/interfaces/IAxelarGasService.sol';
+
+contract Hello is IAxelarExecutable {
+
+    string public message;
+    string public sourceChain;
+    string public sourceAddress;
+    IAxelarGasService gasReceiver;
+
+    constructor(address _gateway, address _gasReceiver) IAxelarExecutable(_gateway) {
+        gasReceiver = IAxelarGasService(_gasReceiver);
+    }
+
+    function sayHello(
+        string memory destinationChain,
+        string memory destinationAddress,
+        bytes calldata payload,
+        string memory symbol,
+        uint256 amount,
+        string memory finalTokenRecipient,
+        uint256 gasFeeAmount,
+        uint256 gasFeeAmountForReturn
+    ) external {
+        address tokenAddress = gateway.tokenAddresses(symbol);
```

Comment:
is it better to check that `amount >= gasFeeAmount + gasFeeAmountForReturn`?
e.g.

```solidity
require(amount >= gasFeeAmount + gasFeeAmountForReturn, "Insufficient amount");
```

That means we ensure that every success tx has enough amount to pay gas for both legs. Then, we can simplify amount passing logic below. For example, we can remove if  block because it always has enough funds to pay gas. Also, we can just pass `amount - gasFeeAmount` at line 56